core/src/main/java/org/javarosa/core/model/User.java

```diff
@@ -0,0 +1,256 @@
+
+    public int recordId = -1; //record id on device
+    public String username;
+    public String password;
+    public String uniqueId;  //globally-unique id
+
+    static private User demo_user;
+
+    public boolean rememberMe = false;
+
+    public String syncToken;
+
+    public byte[] wrappedKey;
+
+    public Hashtable<String, String> properties = new Hashtable<String, String>();
+
+    public User() {
+        setUserType(STANDARD);
+    }
+
+    public User(String name, String passw, String uniqueID) {
+        this(name, passw, uniqueID, STANDARD);
+    }
+
+    public User(String name, String passw, String uniqueID, String userType) {
+        username = name;
+        password = passw;
+        uniqueId = uniqueID;
+        setUserType(userType);
+        rememberMe = false;
+    }
+
+    // fetch the value for the default user and password from the RMS
+    public void readExternal(DataInputStream in, PrototypeFactory pf) throws IOException, DeserializationException {
+        this.username = ExtUtil.readString(in);
+        this.password = ExtUtil.readString(in);
+        this.recordId = ExtUtil.readInt(in);
+        this.uniqueId = ExtUtil.nullIfEmpty(ExtUtil.readString(in));
+        this.rememberMe = ExtUtil.readBool(in);
+        this.syncToken = ExtUtil.nullIfEmpty(ExtUtil.readString(in));
+        this.properties = (Hashtable)ExtUtil.read(in, new ExtWrapMap(String.class, String.class), pf);
+        this.wrappedKey = ExtUtil.nullIfEmpty(ExtUtil.readBytes(in));
+    }
+
+    public void writeExternal(DataOutputStream out) throws IOException {
+        ExtUtil.writeString(out, username);
+        ExtUtil.writeString(out, password);
+        ExtUtil.writeNumeric(out, recordId);
+        ExtUtil.writeString(out, ExtUtil.emptyIfNull(uniqueId));
+        ExtUtil.writeBool(out, rememberMe);
+        ExtUtil.writeString(out, ExtUtil.emptyIfNull(syncToken));
+        ExtUtil.write(out, new ExtWrapMap(properties));
+        ExtUtil.writeBytes(out, ExtUtil.emptyIfNull(wrappedKey));
+    }
+
+    public boolean isAdminUser() {
+        return ADMINUSER.equals(getUserType());
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setID(int recordId) {
+
+        this.recordId = recordId;
+    }
+
+    public int getID() {
+        return recordId;
+    }
+
+    public String getUserType() {
+        if (properties.containsKey(KEY_USER_TYPE)) {
+            return properties.get(KEY_USER_TYPE);
+        } else {
+            return null;
+        }
+    }
+
+    public void setUserType(String userType) {
+        properties.put(KEY_USER_TYPE, userType);
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public boolean isRememberMe() {
```

Comment:
@ctsims can we remove the rememberMe boolean and its getters/setters entirely? It does seem basically unused throughout the codebase